File: lachancuocgoi_kotline/app/src/main/java/com/example/lachancuocgoi/Analysis/AnalysisLevel.kt
package com.example.lachancuocgoi.Analysis

/**
 * Type-safe representation of analysis engine level.
 * Thay thế magic strings ("L1", "L2", "L2-AI", "L3") bằng sealed class
 * để compiler bắt lỗi thiếu case trong when-expression.
 */
sealed class AnalysisLevel {
    /** L1: Keyword Trie + Aho-Corasick matching */
    object L1 : AnalysisLevel() {
        override fun toString() = "L1"
    }

    /** L2: GDetection Engine (keyword + pattern + scenario) */
    object L2 : AnalysisLevel() {
        override fun toString() = "L2"
    }

    /** L2-AI: GDetection với Intent Classification (TFLite) */
    object L2AI : AnalysisLevel() {
        override fun toString() = "L2-AI"
    }

    /** L2-Fused: GDetection + WFSA kết hợp */
    object L2Fused : AnalysisLevel() {
        override fun toString() = "L2-Fused"
    }

    /** L3: Gemini API analysis */
    object L3 : AnalysisLevel() {
        override fun toString() = "L3"
    }

    companion object {
        /** Parse từ string (cho backward compatibility với saved data) */
        fun fromString(value: String): AnalysisLevel = when {
            value == "L1" -> L1
            value == "L2-AI" -> L2AI
            value == "L2-Fused" -> L2Fused
            value.startsWith("L2") -> L2
            value == "L3" -> L3
            else -> L1
        }
    }
}

File: lachancuoicgoi_flutter_demo/lib/analysis/analysis_level.dart
enum AnalysisLevel {
  l1('L1', 'Cấp 1'),
  l2('L2', 'Cấp 2'),
  l2Ai('L2-AI', 'Cấp 2 AI'),
  l2Fused('L2-Fused', 'Cấp 2 hợp nhất'),
  l3('L3', 'Cấp 3 Gemini');

  const AnalysisLevel(this.id, this.displayName);

  final String id;
  final String displayName;

  static AnalysisLevel fromId(String? value) {
    final normalized = value?.trim().toUpperCase();
    return switch (normalized) {
      'L1' => AnalysisLevel.l1,
      'L2' => AnalysisLevel.l2,
      'L2AI' || 'L2-AI' => AnalysisLevel.l2Ai,
      'L2FUSED' || 'L2-FUSED' => AnalysisLevel.l2Fused,
      'L3' => AnalysisLevel.l3,
      _ => AnalysisLevel.l1,
    };
  }
}
